### PR TITLE
Add referrer details task

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

[Trello ticket](https://trello.com/c/ImdExPQ8/679-referrer-details)

# Changes in this PR

This PR introduces the 'Add your details' task for referrer details.

I'm not sure how likely it is that we will have Nomis users without email addresses, and at the moment we're doing nothing to handle this possibility, beyond the generic warning message in case of incorrect details. Maybe this is enough 🤷 but anyway thoughts welcome!

## Screenshots of UI changes

Screenshot 2023-12-15 at 11 09 12

Screenshot 2023-12-15 at 11 10 51

Screenshot 2023-12-15 at 11 09 59

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by patrickjfl** [comment]
[Referrer details](https://trello.com/c/ImdExPQ8/679-referrer-details)

File: integration_tests/fixtures/applicationData.json [hunk 1]
```diff
@@ -13,6 +13,11 @@
       "consentDate-day": "1"
     }
   },
+  "referrer-details": {
+    "confirm-details": { "name": "Eric Dier", "email": "eric.dier@moj.gov.uk" },
```
Comment (Harriethw) [inline integration_tests/fixtures/applicationData.json:17]: I don't think we are saving the name/email on the application document? Looks like from CYA page it's just the job title and contact number?

**Review by Harriethw — APPROVED** [review]
Thanks Pat, this looks good.
I don't think we need to worry at this point about users without email addresses...maybe @edavey would know where/who we could ask to get another report of the percentage of users without an email?
I've created a ticket to add the telephone number to the 'Application summary' here https://trello.com/c/qIkHYkt6/734-referrer-details-telephone-number-as-first-class-field

File: integration_tests/fixtures/applicationData.json (hunk 1 above)
Comment (patrickjfl) [reply integration_tests/fixtures/applicationData.json:17]: aha, good spot! I've added them in as hidden fields in the Nunjucks template.

Comment (Harriethw) [reply integration_tests/fixtures/applicationData.json:17]: oh no, sorry it's right that we aren't saving them! We don't want two sources of truth...  I think?

**Pull request merged**